commit db1312dd95488b5e6ff362ff66fcf953a46b1821
Author: Hannes Reinecke <hare@suse.de>
Date:   Mon Jun 27 11:52:05 2022 +0200

    nvmet: implement basic In-Band Authentication
    
    Implement NVMe-oF In-Band authentication according to NVMe TPAR 8006.
    This patch adds three additional configfs entries 'dhchap_key',
    'dhchap_ctrl_key', and 'dhchap_hash' to the 'host' configfs directory.
    The 'dhchap_key' and 'dhchap_ctrl_key' entries need to be in the ASCII
    format as specified in NVMe Base Specification v2.0 section 8.13.5.8
    'Secret representation'.
    'dhchap_hash' defaults to 'hmac(sha256)', and can be written to to
    switch to a different HMAC algorithm.
    
    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/drivers/nvme/target/admin-cmd.c b/drivers/nvme/target/admin-cmd.c
index 31df40ac828f..fc8a957fad0a 100644
--- a/drivers/nvme/target/admin-cmd.c
+++ b/drivers/nvme/target/admin-cmd.c
@@ -1018,6 +1018,8 @@ u16 nvmet_parse_admin_cmd(struct nvmet_req *req)
 
 	if (nvme_is_fabrics(cmd))
 		return nvmet_parse_fabrics_admin_cmd(req);
+	if (unlikely(!nvmet_check_auth_status(req)))
+		return NVME_SC_AUTH_REQUIRED | NVME_SC_DNR;
 	if (nvmet_is_disc_subsys(nvmet_req_subsys(req)))
 		return nvmet_parse_discovery_cmd(req);